### ISSUE TITLE
Move AbstractConnector's XML property to the containing component

### ISSUE THREAD

**Issue by chrisdembia** [opensim-org/opensim-core#1009]
This issue is to record a discussion I had with @aseth1 about a possibly large design change to the way Connectors and Inputs record their connectees. The idea is for the `connectee_name` property to live in the containing component rather than within the Input or Connector. We would not serialize Inputs or Connectors themselves, but they would get an associated property in the containing component. We could name that property something like `input_<name>_connectee_name`. Here's an example of the change in XML:

Before:

``` xml
<MyComponent>
  <connectors>
    <Connector_PhysicalFrame_ name="parent_frame">
      <connectee_name>../ground</connectee_name>
    </Connector_PhysicalFrame_>
  </connectors>
</MyComponent>
```

After:

``` xml
<MyComponent>
  <!-- Single-value Connector to PhysicalFrame. -->
  <connector_parent_frame_connectee_name>../ground</connector_parent_frame_connectee_name>
</MyComponent>
```

@aseth1 also mentioned that we could potentially view this property as the "default" connectees, which could be modified in-memory after loading the model.

Pros:
1. No need to register Connector and Input objects. In fact, they would not need to derive from `Object` anymore, and could potentially be nested in `Component`.
2. No error-prone xml format, where right now users could edit the names of connectors, their types, and could introduce new connectors that don't actually exist in the component.
3. Fixes an issue we have with trying to remember if an Input is a list input or a single-value input.
4. Less verbose way to express connectors in XML (#359).
5. We could automatically generate the comment for the xml property so that it mentions if the connector is a list connector, and the type that it connects to.

Cons:
1. The simplest solution would make the new property (`connector_<name>_connectee_name`) public in the containing class. Right now, the `connectee_name` property is private in `AbstractConnector`. But we still allow people to edit it through accessors/mutators we've created. We must be careful to keep the property in sync with the `_connectees` vector of references.
2. In XML, the inputs and connectors will not be grouped together anymore. Right now, all connectors and inputs show up together at the top of a Component's xml element. With this change, the connectors/inputs would be scattered throughout. This isn't really so bad. Also, we could update the serialization code to serialize connectors before all other properties.
3. What if we learn, down the road, that we actually want more properties associated with a connector other than just its `connectee_name`? Say connectors eventually get more features (e.g., enabled/disabled). Then we would need to introduce even more properties into the containing component, and this could get messy.

**Comment by tkuchida** [comment]
Cool; Thelen2003Muscle is designed this way. I like hiding the gobbledygook if possible. Perhaps additional properties could be included as XML element attributes:
```xml
<MyComponent>
  <!-- Single-value Connector to PhysicalFrame. -->
  <connector_parent_frame_connectee_name enabled="true" path="../ground" />
</MyComponent>
```

**Comment by chrisdembia** [comment]
That could be neat. We could achieve something similar to that if we created a new `SimpleProperty`, like  `InputConnecteeName`. I don't think it could be quite as compact as you suggest, but we could have:

```xml
<MyComponent>
  <!-- Single-value Input of type double. -->
  <input_desired_angle_connectee_name>
    <InputConnecteeName path="../marker_data" output="column" annotation="right_ankle"/>
  </input_desired_angle_connectee_name>
</MyComponent>
```

**Comment by chrisdembia** [comment]
I just looked into the SimpleProperty code. I'm not so sure that we can use xml attributes to deserialize them :/.

Another issue: right now, properties require one to call `constructProperty_prop_name()` manually. But we can't expect component developers to manually call this method for a property we created within the connector/input macros. So we would need to wait until we use non-static member initializers for properties as well.
